mlflow/tracking/client.py

```diff
@@ -636,7 +636,8 @@ def log_metric(
         Log a metric against the run ID.
 
         :param run_id: The run id to which the metric should be logged.
-        :param key: Metric name.
+        :param key: Metric name (string). This string may only contain alphanumerics, underscores 
```

Comment:
Can we specify the key size that's guaranteed to be supported by all backends for metrics as well?